### PR TITLE
Upgrade ResidualSFX.new_from_existing error handling

### PR DESCRIPTION
**Pull request by Koopa1018**
# Description of changes
Upgrades `ResidualSFX.new_from_existing` to handle more errors, better.

In particular, it now does the following:

- Crashes if the node it was given is not an audio source. (This can be downgraded to an error if needed.)
- Prints an error if the node it was given is an `AudioStreamPlayer3D`. Since the game is 2D, these probably shouldn't be in use anyway.
- Prints an error if the destroy-on-finished-playing signal is connected twice. This situation already errors, but handling it manually lets us give more specific information: what exactly is going wrong, what's been known to cause this before, where to check to try and fix it.

# Issue(s)
Was supposed to deal with #130. It didn't, but it should still be helpful in fixing that bug.

### CODE REVIEW

File: classes/misc/residual_sfx/residual_sfx.gd [hunk 1]
```diff
@@ -24,10 +24,22 @@ func _init(sound: AudioStream, pos: Vector2):
 # Takes an existing AudioStreamPlayer(2D) and makes it act like a ResidualSFX
 # (outlive its parent, optionally destroy on finish).
 static func new_from_existing (
-	sfx,
+	sfx: Node,
 	scene_root: Node,
 	destroy_on_finished = true
 ):
+	# VALIDATE: Is the passed node an audio source?
+	assert(
```
Comment (GTcreyon) [inline classes/misc/residual_sfx/residual_sfx.gd:32]: A crash seems excessive. An error should be fine, no? We're not handling saved data, nor would anything particularly serious occur should this be incorrect.

File: classes/misc/residual_sfx/residual_sfx.gd [hunk 2]
```diff
@@ -24,10 +24,22 @@ func _init(sound: AudioStream, pos: Vector2):
...
+		sfx is AudioStreamPlayer or sfx is AudioStreamPlayer2D \
+			or sfx is AudioStreamPlayer3D, # just for completeness
```
Comment (GTcreyon) [inline classes/misc/residual_sfx/residual_sfx.gd:34]: Please capitalize comments.

**Review by GTcreyon — CHANGES_REQUESTED** [review]
Overall good, but I think demoting to an error rather than a crash seems best.

**Comment by Koopa1018** [comment]
Arright, fixed.

**Review by GTcreyon — APPROVED** [review]

**Pull request merged**